jupyterhub_moss/utils.py

```diff
@@ -44,6 +44,26 @@ def parse_timelimit(timelimit: str) -> datetime.timedelta:
     )
 
 
+_GPU_GRES_REGEXP = re.compile(r"^gpu:(?P<type_>[^:]+):(?P<count>[0-9]+)(?:\(.*\))?$")
+
+
+def parse_gpu_resource(generic_resources: str) -> tuple[str, str]:
```

Comment:
Careful, the `tuple[str, str]` syntax (instead of `Tuple[str, str]`) is only supported for Python 3.9+

Reply:
It's supported from python3.7 thanks to `from __future__ import annotations`

Reply:
Oh right, didn't see the import :+1: